packages/rehype-mathjax/lib/core.js

```diff
@@ -0,0 +1,47 @@
+const visit = require('unist-util-visit')
+
+module.exports = createPlugin
+
+function createPlugin(displayName, createRenderer) {
+  attacher.displayName = displayName
+
+  return attacher
+
+  function attacher(options) {
+    const renderer = createRenderer(options)
+
+    transform.displayName = displayName + 'Transform'
+
+    return transform
+
+    function transform(tree) {
+      let context = tree
```

Comment:
- Is `tree` hast syntax tree that represents either complete HTML document `<html><head>...</head><body>...</body></html>` or a fragment of HTML document such as `<h1>...</h1><p>...</p>` ? 
- If `tree` is a complete HTML document, this plugin inserts stylesheet into `<head>` element.

Reply:
That is correct! It could be any set of elements: a complete document, or not. What this indeed does is, append to head if there is one, and otherwise append to the whole thing!

Reply:
Sounds good. Thank you.